cmd/osbuild-worker/main.go

```diff
@@ -266,10 +275,16 @@ func main() {
 			}
 		}
 
+		proxy := ""
+		if config.Composer != nil && config.Composer.Proxy != "" {
+			proxy = config.Composer.Proxy
```

Comment:
This could be done outside the conditional, and we could parse the proxy into an URL here rather than in the client constructor. That way catching syntax errors in `cmd/`, which is where I think it belongs.

Reply:
Agreed, the worker code would really benefit from some refactoring, let's thinking about it in the future?

Reply:
Let's!